great_expectations/render/renderer_configuration.py

```diff
@@ -274,27 +275,30 @@ class _RendererParamBase(_RendererValueBase):
             but it is dynamically renamed in order for the RendererParams attribute to have the same name as the param.
         """  # noqa: E501
 
-        renderer_schema: RendererSchema = Field(alias="schema")
-        value: Any
-        suite_parameter: Optional[Dict[str, Any]]
+        renderer_schema: RendererSchema = Field(alias="schema", allow_mutation=False)
+        value: Any = Field(allow_mutation=False)
+        suite_parameter: Optional[Dict[str, Any]] = Field(allow_mutation=False)
+        render_state: Optional[ObservedValueRenderState] = Field(allow_mutation=True)
 
         class Config:
             validate_assignment = True
             arbitrary_types_allowed = True
-            allow_mutation = False
 
         @root_validator(pre=True)
         def _validate_param_type_matches_value(cls, values: dict) -> dict:
             """
             This root_validator ensures that a value can be parsed by its RendererValueType.
             If RendererValueType.OBJECT is passed, it is treated as valid for any value.
             """
-            param_type: RendererValueType = values["schema"]["type"]
-            value: Any = values["value"]
-
-            (tester, error) = PARAM_TYPE_TESTERS[param_type]
-            if not tester(value):
-                raise RendererConfigurationError(error(param_type, value))
+            # if render_state is in values, this root_validator is being called
+            # on assignment rather than during instantiation
+            if not values.get("render_state"):
```

Comment:
It's a very annoying pydanic v1 issue where the `root_validator` is run on assignment. Because we have the aliased attribute `renderer_schema` -> `schema` it gets passed in as `schema` upon instantiation and then internally it becomes `renderer_schema`. Then this `root_validator` runs again when we assign to `render_state` and the value in this dict is now `renderer_schema` and there is a `KeyError` that `schema` cannot be found. The alternative would be to check for `schema` or `renderer_schema`, but I felt like it shouldn't run any of this logic again regardless. I think the issue is fixed in pydantic v2 and the analog to `root_validator` isn't called after instantiation.